src/Cardano/Wallet/Primitive.hs

```diff
@@ -221,6 +223,15 @@ newtype Address = Address
 
 instance NFData Address
 
+data AddressError
+    = AddressDecodeError T.Text
```

Comment:
What about a record with meaningful field to give better meaning to the underlying `Text` (also, the qualified import isn't necessary here).

Reply:
Removed - this type was moved to different module (by other author) https://github.com/input-output-hk/cardano-wallet/blob/5c7d690e5d994c931e5b437c5bdbaf222f6dc675/src/Cardano/Wallet/Api/Types.hs#L205 .

It is not a record but it also doesn't seem to be used as a sum type in future. Rather it looks like a self containing wrapper type where constructor/type name tells the storry (at least for now). So I think we can forget about this until we run into more complicated use case